enterprise/server/backends/distributed/distributed.go

```diff
@@ -468,22 +471,60 @@ func (c *Cache) lookasideCacheEnabled() bool {
 	return c.config.LookasideCacheSizeBytes > 0
 }
 
-func (c *Cache) teeReadCloser(r *rspb.ResourceName, rc io.ReadCloser) io.ReadCloser {
-	if !c.lookasideCacheEnabled() {
-		return rc
+func (c *Cache) localReadthroughEnabled() bool {
+	return c.config.ReadThroughLocalCache
+}
+
+func combineCommittedWriteClosers(a, b interfaces.CommittedWriteCloser) interfaces.CommittedWriteCloser {
+	if a == nil {
+		return b
 	}
-	if r.GetDigest().GetSizeBytes() > *maxLookasideEntryBytes {
-		return rc
+	if b == nil {
+		return a
 	}
-	k, ok := lookasideKey(r)
-	if !ok {
+
+	c := io.MultiWriter(a, b)
+	cwc := ioutil.NewCustomCommitWriteCloser(c)
+	cwc.CommitFn = func(n int64) error {
+		if err := a.Commit(); err != nil {
+			return err
+		}
+		return b.Commit()
+	}
+	cwc.CloseFn = func() error {
+		a.Close()
+		b.Close()
```

Comment:
This swallows both of the errors from the delegated `Close()` calls -- is that what you want?

Reply:
Added better logging in a couple places -- we now return this error and log it at INFO level if it happens. I think these will be not-uncommon as they can happen if a Read is closed partway through.